src/main/kotlin/com/team4099/robot2022/subsystems/Wheel.kt

```diff
@@ -25,31 +27,31 @@ import com.team4099.lib.units.derived.radians
 import com.team4099.lib.units.derived.volts
 import com.team4099.lib.units.inFeetPerSecond
 import com.team4099.lib.units.perSecond
-import com.team4099.lib.units.sparkMaxAngularMechanismSensor
-import com.team4099.lib.units.sparkMaxLinearMechanismSensor
 import com.team4099.robot2022.config.Constants
 import edu.wpi.first.math.filter.MedianFilter
 import kotlin.math.IEEErem
 import kotlin.math.sign
 import kotlin.math.withSign
 
 class Wheel(
-  private val directionSpark: CANSparkMax,
-  private val driveSpark: CANSparkMax,
+  private val directionFalcon: TalonFX,
+  private val driveFalcon: TalonFX,
   private val encoder: CANCoder,
   private val zeroOffset: Angle,
   val label: String
 ) {
 
-  private val directionPID = directionSpark.pidController
-  private val drivePID = driveSpark.pidController
-
   private val directionSensor =
-      sparkMaxAngularMechanismSensor(
-          directionSpark, Constants.Drivetrain.DIRECTION_SENSOR_GEAR_RATIO)
+      ctreAngularMechanismSensor(
+          directionFalcon,
+          Constants.Drivetrain.DIRECTION_SENSOR_CPR,
+          Constants.Drivetrain.DIRECTION_SENSOR_GEAR_RATIO)
   private val driveSensor =
-      sparkMaxLinearMechanismSensor(
-          driveSpark, Constants.Drivetrain.DRIVE_SENSOR_GEAR_RATIO, 3.inches)
+      ctreLinearMechanismSensor(
+          driveFalcon,
+          Constants.Drivetrain.DRIVE_SENSOR_CPR,
+          Constants.Drivetrain.DRIVE_SENSOR_GEAR_RATIO,
+          3.inches)
 
   private val directionAbsolute =
       AngularMechanismSensor(
```

Comment:
so the stuff you feed this should also change when you change to the absolute encoder. analog potentiometers cannot read velocity (but we don't care about that so it doesn't matter)